src/back/TicketManager/tests/PaymentService/TicketManager.PaymentService.ServicesTests/Endpoints/Payments/ConfirmPaymentEndpointTest.cs

```diff
@@ -0,0 +1,77 @@
+using FastEndpoints;
+using FluentAssertions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.EntityFrameworkCore;
+using Moq;
+using TicketManager.PaymentService.Contracts.Payments;
+using TicketManager.PaymentService.Domain.Payments;
+using TicketManager.PaymentService.Services.DataAccess;
+using TicketManager.PaymentService.Services.DataAccess.Repositories;
+using TicketManager.PaymentService.Services.Endpoints.Payments;
+using Xunit;
+
+namespace TicketManager.PaymentService.ServicesTests.Endpoints.Payments;
+
+public class ConfirmPaymentEndpointTest
+{
+    private readonly ConfirmPaymentEndpoint endpoint;
+    private readonly Mock<Payment> paymentMock;
+    private readonly Payment payment;
+    private readonly Mock<Repository<Payment, Guid>> paymentsMock;
+    
+    public ConfirmPaymentEndpointTest()
+    {
+        paymentMock = new Mock<Payment>();
+        payment = paymentMock.Object;
+        var dbContextMock = new Mock<PaymentDbContext>(new DbContextOptionsBuilder<PaymentDbContext>().Options);
+        var dbContext = dbContextMock.Object;
+        paymentsMock = new Mock<Repository<Payment, Guid>>(dbContext);
+        paymentsMock.Setup(p => p.FindAndEnsureExistenceAsync(payment.Id, default)).ReturnsAsync(payment);
+        var payments = paymentsMock.Object;
+        endpoint = Factory.Create<ConfirmPaymentEndpoint>(payments);
+    }
+    
+    [Fact]
+    public async Task WhenItIsCalledAndIsSuccessful_ItShouldConfirmPayment()
+    {
+        paymentMock.Setup(p => p.ConfirmPayment()).Returns(true);
+        
+        await endpoint.HandleAsync(new ConfirmPaymentRequest { Id = payment.Id}, default);
+        
+        paymentsMock.Verify(e => e.FindAndEnsureExistenceAsync(payment.Id, default), Times.Once);
+        paymentMock.Verify(e => e.ConfirmPayment(), Times.Once);
+        paymentsMock.Verify(e => e.Update(It.IsAny<Payment>()), Times.Once);
+        
+        endpoint.HttpContext.Response.StatusCode.Should().Be(StatusCodes.Status200OK);
```

Comment:
You mixed validation and logic in something that is meant to implement logic and now it's almost untestable.

Reply:
Could you please elaborate what do you mean with "untestable" and "mixed validation logic"? Is it an additional remark to the previous comments about validators or another concern? Right now this comment doesn't really help me understand the issue.

Reply:
You have to do some hacks inside tests. If you divide validation and logic to separate modules, you can test them independently.

Reply:
Addressed in 7f75126c.